src/praisonai-agents/praisonaiagents/tools/registry.py

```
@@ -184,6 +184,38 @@ def list_available_tools(self, context: Optional[Dict[str, Any]] = None) -> List
+            # Get all available tool names
+            all_tools = self.list_tools()
+            
+            # Apply HERMES_ONLY_TOOLS filter with diagnostics
+            filtered_tools = filter_tools_with_hermes(all_tools, log_diagnostics=True)
+            
+            return sorted(filtered_tools)
+            
+        except ImportError:
+            logging.warning("HERMES_ONLY_TOOLS filter not available, returning all tools")
+            return self.list_tools()
+        except Exception as e:
```

Ruff (0.15.13)

[warning] Do not catch blind exception: `Exception`

(BLE001)